cmd/erasure-readfile.go

```diff
@@ -328,14 +340,23 @@ func decodeData(enBlocks [][]byte, dataBlocks, parityBlocks int) error {
 		return traceError(err)
 	}
 
-	// Verify reconstructed blocks (parity).
+	// Verify correctness of all parity blocks.
 	ok, err := rs.Verify(enBlocks)
 	if err != nil {
 		return traceError(err)
 	}
 	if !ok {
-		// Blocks cannot be reconstructed, corrupted data.
-		err = errors.New("Verification failed after reconstruction, data likely corrupted")
+		// Verification of the parity blocks failed, this means one of two things:
+		// - either one (or more) of the data blocks is corrupted,
+		// - or one (or more) of the parity blocks is corrupted.
+		//
+		// Unfortunately it is not possible to state which one of the two possibilities
+		// it is, so while the data may still be intact, we have no way of saying this
+		// definitively.
+		//
+		// Note that in the presence of bit-rot protection (as is present in Minio)
+		// you really should never see this message.
+		err = errors.New("Unable to verify parity blocks, either existing data or parity blocks corrupted")
```

Comment:
Discuss this and decided to remove the Verification altogether since in our case we are protected by bit rot protection.